src/lib/connections.ts

```diff
@@ -43,26 +45,44 @@ export async function submitAddControlKey(
   signingAccount: Account,
   msaId: number
 ) {
+  if (!api || !(await api.isReady)) {
+    console.debug('api is not available.');
+    return;
+  }
+
   const blockNumber = (await getBlockNumber(api)) as bigint;
-  if (api && (await api.isReady)) {
-    const rawPayload: AddKeyData = {
-      msaId: msaId.toString(),
-      expiration: (blockNumber + 100n).toString(),
-      newPublicKey: newAccount.address,
-    };
 
-    const newKeyPayload = api.registry.createType('PalletMsaAddKeyData', rawPayload);
+  const rawPayload: AddKeyData = {
+    msaId: msaId.toString(),
+    expiration: (blockNumber + 100n).toString(),
+    newPublicKey: newAccount.address,
+  };
 
-    const ownerKeySignature = await signPayload(newKeyPayload, signingAccount, extension);
-    const newKeySignature = await signPayload(newKeyPayload, newAccount, extension);
+  const newKeyPayload = api.registry.createType('PalletMsaAddKeyData', rawPayload);
 
-    const ownerKeyProof = { Sr25519: ownerKeySignature };
-    const newKeyProof = { Sr25519: newKeySignature };
-    const extrinsic = api.tx.msa.addPublicKeyToMsa(signingAccount.address, ownerKeyProof, newKeyProof, newKeyPayload);
-    submitExtinsic(extrinsic, signingAccount, extension);
-  } else {
-    console.debug('api is not available.');
+  // mock signatures for fee estimation
+  const mockSignature = '0x' + '00'.repeat(64);
+  const mockProof = { Sr25519: mockSignature };
+  // mock extrinsic for fee estimation
+  const mockExtrinsic = api.tx.msa.addPublicKeyToMsa(signingAccount.address, mockProof, mockProof, newKeyPayload);
+  // typecheck for testing
+  if (typeof mockExtrinsic.paymentInfo === 'function') {
+    // Get estimated total cost of txn & user's transferable balance
+    const estTotalCost = (await mockExtrinsic.paymentInfo(signingAccount.address)).partialFee.toBigInt();
+    const transferable = BigInt(get(user).balances.transferable);
+    // Check for adiquite funds
+    if (transferable < estTotalCost) throw new Error('User does not have sufficient funds.');
```

Comment:
Does this check take into account the ED?

Also, typo:
```suggestion
    const transferable = BigInt(get(user).balances.transferable);
    // Check for adequate funds
    if (transferable < estTotalCost) throw new Error('User does not have sufficient funds.');
```

Reply:
Existential Deposit. Basically a minimum balance enforced on a token account.